packages/block-editor/src/components/use-display-block-controls/index.js

```diff
@@ -1,36 +1,8 @@
-/**
- * WordPress dependencies
- */
-import { useSelect } from '@wordpress/data';
-
 /**
  * Internal dependencies
  */
 import { useBlockEditContext } from '../block-edit/context';
-import { store as blockEditorStore } from '../../store';
 
 export default function useDisplayBlockControls() {
-	const { isSelected, clientId, name } = useBlockEditContext();
-	return useSelect(
-		( select ) => {
-			if ( isSelected ) {
-				return true;
-			}
-
-			const {
-				getBlockName,
-				isFirstMultiSelectedBlock,
-				getMultiSelectedBlockClientIds,
-			} = select( blockEditorStore );
-
-			if ( isFirstMultiSelectedBlock( clientId ) ) {
-				return getMultiSelectedBlockClientIds().every(
-					( id ) => getBlockName( id ) === name
-				);
-			}
-
-			return false;
-		},
-		[ clientId, isSelected, name ]
-	);
+	return useBlockEditContext().shouldDisplayControls;
```

Comment:
`useBlockControlsFill` calls `useDisplayBlockControls` but also adds some additional checks that are now part of `useDisplayBlockControls` so there's probably some code to remove from `useBlockControlsFill`